pkg/reconciler/v1alpha1/clusteringress/clusteringress.go

```diff
@@ -0,0 +1,205 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusteringress
+
+import (
+	"context"
+	"reflect"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/knative/pkg/apis/istio/v1alpha3"
+	istioinformers "github.com/knative/pkg/client/informers/externalversions/istio/v1alpha3"
+	istiolisters "github.com/knative/pkg/client/listers/istio/v1alpha3"
+	"github.com/knative/pkg/controller"
+	"github.com/knative/pkg/logging"
+	"github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	informers "github.com/knative/serving/pkg/client/informers/externalversions/networking/v1alpha1"
+	listers "github.com/knative/serving/pkg/client/listers/networking/v1alpha1"
+	"github.com/knative/serving/pkg/reconciler"
+	"github.com/knative/serving/pkg/reconciler/v1alpha1/clusteringress/resources"
+	"github.com/knative/serving/pkg/reconciler/v1alpha1/clusteringress/resources/names"
+)
+
+const controllerAgentName = "clusteringress-controller"
+
+// Reconciler implements controller.Reconciler for ClusterIngress resources.
+type Reconciler struct {
+	*reconciler.Base
+
+	// listers index properties about resources
+	clusterIngressLister listers.ClusterIngressLister
+	virtualServiceLister istiolisters.VirtualServiceLister
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// NewController initializes the controller and is called by the generated code
+// Registers eventhandlers to enqueue events.
+func NewController(
+	opt reconciler.Options,
+	clusterIngressInformer informers.ClusterIngressInformer,
+	virtualServiceInformer istioinformers.VirtualServiceInformer,
+) *controller.Impl {
+
+	c := &Reconciler{
+		Base:                 reconciler.NewBase(opt, controllerAgentName),
+		clusterIngressLister: clusterIngressInformer.Lister(),
+		virtualServiceLister: virtualServiceInformer.Lister(),
+	}
+	impl := controller.NewImpl(c, c.Logger, "ClusterIngresses")
+
+	c.Logger.Info("Setting up event handlers")
+	clusterIngressInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    impl.Enqueue,
+		UpdateFunc: controller.PassNew(impl.Enqueue),
+		DeleteFunc: impl.Enqueue,
+	})
+
+	virtualServiceInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("ClusterIngress")),
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc:    impl.EnqueueControllerOf,
+			UpdateFunc: controller.PassNew(impl.EnqueueControllerOf),
```

Comment:
There isn't enough information in OwnerReference for this to work properly.

See also: https://github.com/knative/serving/issues/1259

Reply:
Thanks for reminding this.
Will construct a selection func instead, via label selector I think.